Orange/data/table.py

```diff
@@ -112,26 +112,40 @@ def __setitem__(self, key, value):
                 self.table.metas[self.row_index, -1 - key] = value
 
     def _str(self, limit):
-        def sp_values(matrix, variables):
+        def sp_values(matrix, variables, show_labels=True):
             if not sp.issparse(matrix):
+                if matrix.ndim == 1:
+                    matrix = matrix[:, np.newaxis]
                 return Instance.str_values(matrix[row], variables, limit)
-            begptr, endptr = matrix.indptr[row:row + 2]
-            rendptr = endptr if not limit else min(endptr, begptr + 5)
-            variables = [variables[var]
-                         for var in matrix.indices[begptr:rendptr]]
-            s = ", ".join(
-                "{}={}".format(var.name, var.str_val(val))
-                for var, val in zip(variables, matrix.data[begptr:rendptr]))
-            if limit and rendptr != endptr:
+
+            # Sparse matrices can't handle slices where an index is out of
+            # bounds like numpy arrays can, so we must determine largest index
+            _, columns = matrix.shape
+            max_idx, has_more_columns = min(5, columns), columns > 5
+
+            row_entries = []
+            for idx, var in enumerate(variables[:max_idx]):
+                s = ""
+                if var.is_discrete or matrix[row, idx]:
+                    if show_labels:
+                        s += "%s=" % var.name
+                    s += var.str_val(matrix[row, idx])
```

Comment:
`s = "%s=" % var.name if show_labels else ''`? Though this might change due to comments above.